quickstarts/microsoft.dataprotection/backup-create-storage-account-enable-protection/main.bicep

```diff
@@ -84,19 +272,28 @@ resource storageAccount 'Microsoft.Storage/storageAccounts@2021-06-01' = {
   }
 }
 
-resource roleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
-  name: guid(backupVault.id, roleDefinitionId, storageAccount.id)
+resource storageContainerList 'Microsoft.Storage/storageAccounts/blobServices/containers@2023-01-01' = [
+  for item in containerList: {
+    name: '${storageAccountName}/default/${item}'
+    dependsOn: [
+      storageAccount
+    ]
+  }
+]
+
+resource roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
   scope: storageAccount
+  name: guid(vault.id, roleDefinitionId, storageAccount.id)
   properties: {
     roleDefinitionId: roleDefinitionId
-    principalId: reference(backupVault.id, '2021-01-01', 'Full').identity.principalId
+    principalId: reference(vault.id, '2021-01-01', 'Full').identity.principalId
     principalType: 'ServicePrincipal'
   }
 }
 
-resource backupInstance 'Microsoft.DataProtection/backupvaults/backupInstances@2021-01-01' = {
-  parent: backupVault
-  name: storageAccountName
+resource backupInstance 'Microsoft.DataProtection/backupVaults/backupInstances@2022-05-01' = {
+  parent: vault
+  name: '${storageAccountName}'
```

Comment:
remove string interpolation

Reply:
Thanks @alex-frankel for the review! Addressed all the review comments. Kindly review again and see if we can merge it soon